### PR TITLE
Adding new TC number 26 "Verify Scroll Up without using 'Arrow' butto…

### PR DESCRIPTION
**Pull request by MostafaAbotalebGizaSystems**
Adding new TC number 26 "Verify Scroll Up without using 'Arrow' button and Scroll Down functionality":

- I added a new test class "VerifyingScrollUpWithoutArrowButtonAndScrollDownTests"
- I added one action method "scrollToTopPage" in the "HomePage"

### CODE REVIEW

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.gizasystems.automationexercise.pages.HomePage;
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Description;
+import io.qameta.allure.TmsLink;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class VerifyingScrollUpWithoutArrowButtonAndScrollDownTests {
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java:13]: Please add @Epic, @Feature & @Story annotations here.

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                .verifySubscriptionText(testUser.getTestData("subText"));
+        new HomePage(driver)
+                .scrollToTopPage()
+                .validateOnVisibilityOfCarouselSlideText(testUser.getTestData("slideHeadingText"));
+    }
+
+    //////////////////// Configurations \\\\\\\\\\\\\\\\\\\\
+    @BeforeClass
+    public void beforeClass() {
+        testUser = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/ScrollingTestData.json");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java:34]: We agreed that we should have 1 to 1 relationship between the test classes and test data json files.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 3]
```diff
@@ -34,6 +34,12 @@ public HomePage clickOnScrollUpArrow() {
         return this;
     }
 
+    @Step("Scroll to top page")
+    public HomePage scrollToTopPage() {
+        driver.element().scrollToElement(slideHeading_div);
+        return this;
+    }
+
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:42]: We need to discuss the scrolling tests as some cases require clicking on the arrow itself and the arrow button is not part of the Home Page.

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java
[the remark this answers is not in the document]
Comment (MostafaAbotalebGizaSystems) [reply src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java:1]: Do you mean 2 test classes of scrolling in one package?
or 2 test cases in one class?

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 3 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:42]: @Step("Click on scroll up arrow")
    public HomePage clickOnScrollUpArrow() {
        driver.element().click(scrollUpArrow_button);
        return this;
    }
    
    private final By scrollUpArrow_button = By.xpath("//i[@class='fa fa-angle-up']");
    
   I think you mean this method and the locator of the arrow are not part of the Home page.
   You are right, let's discuss this next meeting

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java
[the remark this answers is not in the document]
Comment (MahmoudElSharkawyGS) [reply src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpWithoutArrowButtonAndScrollDownTests.java:1]: 2 test cases on one test class as they are related.

**Review by MahmoudElSharkawyGS — APPROVED** [review]

**Pull request merged**